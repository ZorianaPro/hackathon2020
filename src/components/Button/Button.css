.Button {
  cursor: pointer;
}

.Button .Button-Inside {

}

.Button p.Button-Label {
  cursor: pointer;
}

.Button.Button--isActionFilled {
  overflow:hidden;
  position: relative;
  width: 506px;
  height: 64px;
}

.Button.Button--isActionFilled p.Button-Label {
  color: white;
  position: absolute;
  left: 50%;
  z-index: 3;
  margin: 0;
  transform: translateX(-30%);
  font-size: var(--fontSize18);
  font-weight: var(--fontWeightPoppinsBold);
  font-style: italic;
  line-height: 64px;
}

.Button.Button--isActionFilled .Button-Inside {
  cursor: pointer;
  transform: rotate(115deg);
  background-color: var(--colorRadicalRed);
  height: 291px;
}

.Button.Button--isActionFilled .Button-Inside:hover {

}

.Button.Button--isActionTransparent {

}

.Button.Button--isActionTransparent .Button-Inside:hover {

}

.Button.Button--isSecondaryTransparent {
  border: solid 1px var(--colorMesiumTorquoise);
  padding: 10px;
  text-align: center;
}

.Button.Button--isSecondaryTransparent .Button-Inside {
  font-weight: var(--fontWeightPoppinsBold);
  font-size: var(--fontSize20);
  display: flex;
  justify-content: center;
  align-items: center;
}

.Button .Button--isSecondaryTransparent .Button-Inside:hover {

}
