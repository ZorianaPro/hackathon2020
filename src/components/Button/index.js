import React from 'react';
import './Button.css'

const Button = ({
	title,
	action = 'overlay',
	target
}) => {

	const clickHandler = () => {
		const targetEl = document.querySelector(`.${target}`);
		if (targetEl) {
			if (action === 'scroll') {
				targetEl.scrollIntoView({ behavior: 'smooth', block: 'start'});
			}
			if (action === 'overlay') {
			//	open target overlay
			}
		}
	};

	return (
		<div className="Button"
		     onClick={() => clickHandler()}>
			{title}
		</div>
	)
};

export default Button;