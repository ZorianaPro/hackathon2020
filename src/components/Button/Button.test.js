import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './';

describe('Button', () => {
  it('renders without crashing', () => {
    expect(shallow.bind(shallow, <Button />)).not.toThrow();
  });

  it('renders as expected', () => {
    expect(shallow(<Button />)).toMatchSnapshot();
  });

  describe('on pass title', () => {
    let button, props;

    beforeEach(() => {
      props = {
        label: 'Any title',
        target: 'Some target',
        action: 'scroll'
      };
      button = mount(<Button {...props} />);
    });

    it('display title', () => {
      expect(button.exists('.Button-Label')).toBe(true);
    });
  });

  describe('on click', () => {
    let button, props;
    const clickHandler = jest.fn();

    beforeEach(() => {
      props = {
        label: 'Any title',
        target: 'Some target',
        action: 'scroll'
      };
      button = mount(<Button {...props} onClick={clickHandler()} />);
      button.simulate('click');
    });

    it('display title', () => {
      expect(clickHandler).toHaveBeenCalledTimes(1);
    });
  });
});
