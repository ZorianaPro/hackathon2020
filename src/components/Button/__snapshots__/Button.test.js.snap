// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders as expected 1`] = `
<div
  className="Button "
>
  <p
    className="Button-Label"
    onClick={[Function]}
  />
  <div
    className="Button-Inside"
    onClick={[Function]}
  />
</div>
`;
